src/widechar_width.h

```diff
@@ -30,8 +30,8 @@ enum {
 
 /* An inclusive range of characters. */
 struct widechar_range {
-  int32_t lo;
```

Comment:
:-1: DO NOT MAKE ANY CHANGES TO THIS FILE - IT IS IMPORTED WHOLESALE FROM UPSTREAM AT: https://github.com/ridiculousfish/widecharwidth AND MAY WELL GET OVERWRITTEN THE NEXT TIME THAT THE UNICODE VERSION IS INCREMENTED.

Reply:
Oh, sorry I was reviewing without looking at any past feedback from anyone else - yes if there are argument/variable type errors then they need to be passed upstream - though remember that this file is designed to work on multiple OSes and so may use some types to ensure compatibility which might not seem to be the optimum ones to us.

The reason that a few pixels may be being chopped off of the right for some Emojis and other *wide* glyphs is, I suspect, because it is not the same font being used for *all* the characters, if the specified font does not contain a wanted glyph then the Qt and Font subsystems (Font Config on Linux for instance) will (normally) work out which other fonts do have the wanted glyph for that grapheme and pull in the best (for *some* value of best) one they can find - however there is no guarantee that it will have the exact same font metrics.  The best way around that would be to measure the actual widths used when the glyphs are painted and to remember them for later when working out where the mouse is click on them - mayhaps I am going to have to dust off that prototype code I cobbled together to do it like that...

Reply:
This file was removed entirely and not used anymore @SlySven.

Reply:
> 👎 DO NOT MAKE ANY CHANGES TO THIS FILE - IT IS IMPORTED WHOLESALE FROM UPSTREAM AT: https://github.com/ridiculousfish/widecharwidth AND MAY WELL GET OVERWRITTEN THE NEXT TIME THAT THE UNICODE VERSION IS INCREMENTED.

I'm not sure whether the problem lies in the library on in some compiling flag for Mudlet.
I'm not so expert with C, but reading here and there seems like `wchar_t` is the correct type to handle 2-bytes or 4-bytes character. But in Mudlet dev environment it is interpreted as a 2-bytes type and cannot handle many unicode ranges.
I'd open an issue to [https://github.com/ridiculousfish/widecharwidth](url) if I'd know for sure.

Reply:
> This file was removed entirely and not used anymore @SlySven.

No it isn't - in fact I recently updated it to support Unicode 12.1 ! :confused:

It's purpose is to tell the caller whether the grapheme is supposed to be drawn in a normal or wide (double the width of a normal) width - taking into account whether the character is defined as a normal or wide one or is an *ambiguous* width or whether it changed from being narrow in Unicode 8 to be wide in Unicode 9. If it is an ambiguous width then our code considers the choice the user made from them - or if they set it to `auto` - the server encoding the user has selected...

The bottom line is that it is trying to approximate the duo-space font we need for MUD type text displays...